src/main/java/beam/physsim/jdeqsim/AgentSimToPhysSimPlanConverter.java

```diff
@@ -138,19 +153,45 @@ public void buildPersonToHousehold() {
                 .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
     }
 
-    private void setupActorsAndRunPhysSim(int iterationNumber) {
-        RelaxationExperiment sim = RelaxationExperiment$.MODULE$.apply(beamConfig, agentSimScenario, jdeqsimPopulation,
-                beamServices, controlerIO, caccVehiclesMap, beamConfigChangesObservable, iterationNumber, rnd);
-        log.info("RelaxationExperiment is {}, type is {}", sim.getClass().getSimpleName(), beamConfig.beam().physsim().relaxation().type());
-        TravelTime travelTimeFromPhysSim = sim.run(prevTravelTime);
-        // Safe travel time to reuse it on the next PhysSim iteration
-        prevTravelTime = travelTimeFromPhysSim;
+    private void setupActorsAndRunPhysSim(IterationEndsEvent iterationEndsEvent) {
+        // I don't use single class `UpdateTravelTime` here and make decision in `BeamRouter` because
+        // below we have `linkStatsGraph.notifyIterationEnds` call which internally will call `BeamCalcLinkStats.addData`
+        // which may change an internal state of travel time calculator (and it happens concurrently in CompletableFuture)
+        //################################################################################################################
+        Collection<? extends Link> links = agentSimScenario.getNetwork().getLinks().values();
+        int maxHour = (int) TimeUnit.SECONDS.toHours(agentSimScenario.getConfig().travelTimeCalculator().getMaxTime()) + 1;
 
-        if (beamConfig.beam().debug().debugEnabled()) {
-            log.info(DebugLib.getMemoryLogMessage("Memory Use After JDEQSim: "));
-        }
+        int iterationNumber = iterationEndsEvent.getIteration();
+
+        Map<String, double[]> travelTimeMap;
+        TravelTime travelTimeFromPhysSim;
+
+        String physSimType = beamConfig.beam().physsim().physSimType();
+
+        if (physSimType.equals("JDEQSim")) {
+            log.info("JDEQSim started");
+
+            RelaxationExperiment sim = RelaxationExperiment$.MODULE$.apply(beamConfig, agentSimScenario, jdeqsimPopulation,
```

Comment:
Should we create a constructor instead of having to use the scala stringification? Do we own this library?